### PR TITLE
remove left_delete handle + handle :,; suffixes

### PR DESCRIPTION
**Pull request by dimacodota**
- remove left_delete handle - it was causing a flickering when deleting
 - handle `,;:` suffixes as stop completion if at the end of the line

### CODE REVIEW

File: completions/completions_v2.py [hunk 1]
```diff
@@ -79,10 +79,10 @@ def _run_complete():
     def should_run_completion_on_modified(self, view):
         current_location = view.sel()[0].end()
         is_disabled = is_tabnine_disabled(view)
-        in_json_end_line = is_json_end_line(view, current_location)
+        stop_completion_after_end_line = should_stop_completion_after_end_line(view, current_location)
         in_query_after_new_line = is_query_after_new_line(
             view, current_location)
-        return current_location - self._last_query_location >= COMPLEATIONS_REQUEST_TRESHOLD and not self._stop_completion and not is_disabled and not in_json_end_line and not in_query_after_new_line
```
Comment (boaz-codota) [inline completions/completions_v2.py:85]: this is insane. Can we break it to multiple if statements that return a false?

File: completions/completions_v2.py [hunk 2]
```diff
@@ -184,10 +184,11 @@ def has_competions(self):
         return len(self._completions) > 0
 
     def handle_tabnine_commands(self, view, locations, prefix):
-        if len(self._results) == 1 and self._old_prefix is None:
+        if len(self._results) == 1 and self._old_prefix is None and prefix != "":
+            seperator = "::"
             existing = view.substr(sublime.Region(
-                max(locations[0] - (len(prefix) + 2), 0), locations[0]))
-            is_tabNine_command = existing == "::{}".format(prefix)
+                max(locations[0] - (len(prefix) + len(seperator)), 0), locations[0]))
```
Comment (boaz-codota) [inline completions/completions_v2.py:190]: you can do the `"{}{}".format(seperator, prefix)` at line 188 and then just use `len(extracted_variable)`

File: lib/view_helpers.py [hunk 3]
```diff
@@ -11,11 +11,10 @@ def get_after(view, char_limit):
     end = min(view.size(), loc + char_limit)
     return view.substr(sublime.Region(loc, end)), end == view.size()
 
-def is_json_end_line(view, current_location):
+def should_stop_completion_after_end_line(view, current_location):
     last_character = view.substr(max(current_location - 1, 0))
     end_of_line = view.line(current_location).end()
-    is_json_end_line = end_of_line == current_location and last_character == ","
-    return is_json_end_line
+    return end_of_line == current_location and last_character in ",;:"
```
Comment (boaz-codota) [inline lib/view_helpers.py:17]: extract ` ",;:"` to a const to give it a name

Comment (boaz-codota) [reply lib/view_helpers.py:17]: plus this is very buggy (think of `lambda x: ` that will not receive suggestions)

**Review by boaz-codota — CHANGES_REQUESTED** [review]
What about some refactoring such as breaking to files and packages? Now seems like a good time to do that...

File: lib/view_helpers.py (hunk 3 above)
Comment (dimacodota) [reply lib/view_helpers.py:17]: yes, you will not get completions right after the `:` only after a whitespace

File: completions/completions_v2.py (hunk 1 above)
Comment (dimacodota) [reply completions/completions_v2.py:85]: 👍

File: completions/completions_v2.py (hunk 2 above)
Comment (dimacodota) [reply completions/completions_v2.py]: 👍

File: completions/completions_v2.py [hunk 4]
```diff
@@ -79,10 +79,22 @@ def _run_complete():
     def should_run_completion_on_modified(self, view):
         current_location = view.sel()[0].end()
         is_disabled = is_tabnine_disabled(view)
-        in_json_end_line = is_json_end_line(view, current_location)
+        stop_completion_after_end_line = should_stop_completion_after_end_line(view, current_location)
         in_query_after_new_line = is_query_after_new_line(
             view, current_location)
-        return current_location - self._last_query_location >= COMPLEATIONS_REQUEST_TRESHOLD and not self._stop_completion and not is_disabled and not in_json_end_line and not in_query_after_new_line
+
+        if current_location - self._last_query_location >= COMPLEATIONS_REQUEST_TRESHOLD:
```
Comment (boaz-codota) [inline completions/completions_v2.py:86]: This is different from what it was. Are we sure about that?

**Review by boaz-codota — APPROVED** [review]

File: completions/completions_v2.py (hunk 4 above)
Comment (dimacodota) [reply completions/completions_v2.py]: fixed 👍

**Pull request merged**